startup_timeout: 10.0

# Message queue item limit
message_queue_item_limit: 3

# Incoming image format
# Can be RGB or RGBA
image_format: BGR

# Settings for the grid position estimation portion of the GridLineEstimator
grid_estimator:
    # Step size for initial orientation sweep
    theta_step: 0.004

    # Step size for initial translation sweep
    grid_step: 0.01

    # Distance between the center of one gridline and the center of the next
    grid_spacing: 0.229

    # Thickness of each gridline
    grid_line_thickness: 0.038

    # (x-location of gridline intersection) - (x-location of origin)
    grid_zero_offset_x: 0.5

    # (y-location of gridline intersection) - (y-location of origin)
    grid_zero_offset_y: 0.5

    # Number of times to iterate in get1dGridShift
    grid_translation_mean_iterations: 1

    # Threshold to reject lines that aren't aligned with the grid
    line_rejection_angle_threshold: 0.262 # pi/12

    # Minimum altitude to run the detector
    min_extraction_altitude: 0.1

    # Allowed lag between available position and frame timestamp
    allowed_position_stamp_error: 0.1

grid_line_estimator:
    # Should we forgo line counting and just detect lines on the ground?
    debug_line_detector: false

    # Should we spit out a marker vector for the quad's estimated orientation?
    debug_direction: false

    # Should we spit out an edge image for each frame on the `edges` topic?
    debug_edges: true

    # Should we spit out a copy of the input with lines drawn on top on the
    # `lines` topic?
    debug_lines: true

    # Should we spit out markers for the transformed lines?
    debug_line_markers: true

    # Uncomment this to override the height from robot_localization
    # debug_height: 0.22

roomba_estimator:
    # Measured in meters
    roomba_plate_height: 0.225
    roomba_plate_width: 0.135
    # Measued in meters
    roomba_height: 0.065

    detection_image_width: 300

    hsv_slice_h_green_min: 60
    hsv_slice_h_green_max: 100
    hsv_slice_s_green_min: 100
    hsv_slice_s_green_max: 255
    hsv_slice_v_green_min: 60
    hsv_slice_v_green_max: 180
    hsv_slice_h_red1_min: 0
    hsv_slice_h_red1_max: 30
    hsv_slice_s_red_min: 130
    hsv_slice_s_red_max: 255
    hsv_slice_v_red_min: 80
    hsv_slice_v_red_max: 200
    hsv_slice_h_red2_min: 160
    hsv_slice_h_red2_max: 180

    min_roomba_blob_size: 100
    max_roomba_blob_size: 15000

    morphology_size: 3
    morphology_iterations: 1

    max_relative_error: 1.0

    # Multiplier on difference between observed and actual plate diagonal to
    # get angle estimate stddev
    uncertainty_scale: 1.0

    # Camera settings
    # https://en.wikipedia.org/wiki/LifeCam#HD-6000

    # Camera diagonal angle of view settings
    #   E-con DFOV: 93.5 deg
    #  Genius DFOV:  120 deg
    #    C920 DFOV:   78 deg
    # Lifecam DFOV:   66 deg
    #     Sim DFOV:   60 deg
    # Leopard DFOV:107.7 deg
    bottom_camera_aov: 107.7

    debug_hsv_slice: true
    debug_contours: true
    debug_detected_rects: true
